chia/full_node/block_store.py

```diff
@@ -616,26 +781,44 @@ async def get_block_records_close_to_peak(
         if peak is None:
             return {}, None
 
-        ret: Dict[bytes32, BlockRecord] = {}
+        ret: Dict[bytes32, BlockRecordDB] = {}
+        all_blocks_plot_filters: Dict[bytes32, PlotFilterInfo] = {}
         if self.db_wrapper.db_version == 2:
             async with self.db_wrapper.reader_no_transaction() as conn:
                 async with conn.execute(
-                    "SELECT header_hash, block_record FROM full_blocks WHERE height >= ?",
+                    "SELECT header_hash, block_record,block FROM full_blocks WHERE height >= ?",
                     (peak[1] - blocks_n,),
                 ) as cursor:
                     for row in await cursor.fetchall():
                         header_hash = bytes32(row[0])
-                        ret[header_hash] = BlockRecord.from_bytes(row[1])
+                        ret[header_hash] = BlockRecordDB.from_bytes(row[1])
+                        all_blocks_plot_filters[header_hash] = plot_filter_info_from_block(zstd.decompress(row[2]))
 
         else:
             formatted_str = f"SELECT header_hash, block  from block_records WHERE height >= {peak[1] - blocks_n}"
             async with self.db_wrapper.reader_no_transaction() as conn:
                 async with conn.execute(formatted_str) as cursor:
                     for row in await cursor.fetchall():
                         header_hash = self.maybe_from_hex(row[0])
-                        ret[header_hash] = BlockRecord.from_bytes(row[1])
+                        ret[header_hash] = BlockRecordDB.from_bytes(row[1])
 
-        return ret, peak[0]
+            formatted_str = f"SELECT header_hash,block  from full_blocks WHERE height >= {peak[1] - blocks_n}"
+            async with self.db_wrapper.reader_no_transaction() as conn:
+                async with conn.execute(formatted_str) as cursor:
+                    for row in await cursor.fetchall():
+                        header_hash = self.maybe_from_hex(row[0])
+                        all_blocks_plot_filters[header_hash] = plot_filter_info_from_block(row[1])
+
+        result: Dict[bytes32, BlockRecord] = {}
+        for hh, block_record_db in ret.items():
```

Comment:
same comment here as above. If you fold this loop into the previous loop(s), I think it will be simpler and faster